es6 & beyond/ch4.md

```diff
@@ -1,26 +1,27 @@
 
-Another way of thinking about a Promise is as an event listener, on which you can register to listen for an event that lets you know when a task has completed. It's an event that will only ever fire once, but it can be thought of as an event nonetheless.
+Outra forma de pensar a respeito de uma Promise é como sendo um event listener, onde você registra um evento que te informa quanto uma tarefa foi concluída. É um evento que será disparado apenas uma vez, mas mesmo assim podemos imaginal-los como se fosse um evento.
 
-Promises can be chained together, which can sequence a series of asychronously completing steps. Together with higher-level abstractions like the `all(..)` method (in classic terms, a "gate") and the `race(..)` method (in classic terms, a "latch"), promise chains provide a mechanism for async flow control.
+Promises podem ser encadeadas, em uma sequencia de passos completados assincronamente. Juntas com uma abstração de alto-nivel como o método `all(...)` (em termo classico, um portal) e o método `race(...)` (em um tempo classico, um trinco), promise encadeadas proporciona um mecanismo para controle de fluxo assincrono.
 
-Yet another way of conceptualizing a Promise is that it's a *future value*, a time-independent container wrapped around a value. This container can be reasoned about identically whether the underlying value is final or not. Observing the resolution of a Promise extracts this value once available. In other words, a Promise is said to be the async version of a sync function's return value.
+Ainda uma outra forma de se entender uma Promise é que tem um *valor futuro*, um recipiente independente de tempo envolvido em torno de um valor. Esse recipiente pode ser .... . Observando a resolução de uma Promise exrtai esse valor uma vez avaliado. Em outras palavras, uma Promise diz ser a versão assincrona de valores retornados por funções.
 
-A Promise can only have one of two possible resolution outcomes: fulfilled or rejected, with an optional single value. If a Promise is fulfilled, the final value is called a fulfillment. If it's rejected, the final value is called a reason (as in, a "reason for rejection"). Promises can only be resolved (fulfillment or rejection) *once*. Any further attempts to fulfill or reject are simply ignored. Thus, once a Promise is resolved, it's an immutable value that cannot be changed.
+Uma promise pode apenas possuir apenas um das duas possíves resoluções: cumprida (fulfilled) ou rejeitada (rejected), com um valor único opicional. Se uma Promise for cumprida, o valor final é chamado de cumprimento. Se for rejeitada, o valor final é denominado de motivo (como sendo, o "motivo da rejeição"). Promises podem apenas ser resolvidas (cumprida ou rejeitada) uma única vez. Qualquer tentativa futura de cumprir ou rejeitar são simplesmente ignoradas. Assim, uma vez que uma Promise for resolvida, seu valor se torna imutavel, o qual não pode ser alterado.
```

Comment:
Uma promise ~~pode apenas~~ possuir apenas um**a,** das duas possíves resoluções: concluída ~~(fulfilled)~~ ou rejeitada ~~(rejected)~~, com um valor único op~~i~~cional. Se uma *Promise* for cumprida, o valor final é chamado de cumprimento. Se for rejeitada, o valor final é denominado de motivo (como sendo, o "motivo da rejeição"). *Promises* podem apenas ser resolvidas (cumprida ou rejeitada) uma única vez. Qualquer tentativa futura de cumpri~~r~~**-la** ou rejeita~~r~~**-la** ~~são~~ **é** simplesmente ignorada~~s~~. Assim, uma vez que uma *Promise* for resolvida, seu valor se torna imut**á**vel, ~~o qual~~ **e** não pode ser alterado.